test/e2e/reboots/stalld.go

```diff
@@ -75,16 +75,19 @@ var _ = ginkgo.Describe("[reboots][stalld] Node Tuning Operator installing syste
 			err = util.WaitForPoolUpdatedMachineCount(cs, "worker-rt", 1)
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 
-			ginkgo.By(fmt.Sprintf("waiting for the tuned daemon running on node %s", node.Name))
-			out, err := util.WaitForCmdInPod(pollInterval, waitDuration, pod, "test", "-e", "/run/tuned/tuned.pid")
-			gomega.Expect(err).NotTo(gomega.HaveOccurred())
-
-			// 10s is very generous to allow for (now unwanted) stalld service starting
-			ginkgo.By("sleeping 10s to allow for (now unwanted) stalld service starting")
-			time.Sleep(10 * time.Second)
+			waitForTuneD := func() {
+				ginkgo.By(fmt.Sprintf("waiting for the TuneD daemon running on node %s", node.Name))
+				_, err := util.WaitForCmdInPod(pollInterval, waitDuration, pod, "test", "-e", "/run/tuned/tuned.pid")
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+				// 10s is very generous to allow for stalld service starting (if any)
+				ginkgo.By("sleeping 10s to allow for stalld service starting (if any)")
+				time.Sleep(10 * time.Second)
```

Comment:
I'm not sure to understand the `(if any)` part of the comment, could you explain it a bit?

what I understand from this function is:

1. wait for file `/run/tuned/tuned.pid` to appear, meaning that TuneD started its (early) execution
2. wait 10s for TuneD to start operating properly